src/test/java/jewellery/inventory/integration/AuthIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+package jewellery.inventory.integration;
+
+import static jewellery.inventory.helper.UserTestHelper.USER_EMAIL;
+import static jewellery.inventory.helper.UserTestHelper.USER_NAME;
+import static jewellery.inventory.helper.UserTestHelper.USER_PASSWORD;
+import static jewellery.inventory.helper.UserTestHelper.createTestUser;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+import jewellery.inventory.dto.request.AuthenticationRequestDto;
+import jewellery.inventory.dto.response.UserAuthDetailsDto;
+import jewellery.inventory.model.User;
+import jewellery.inventory.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthIntegrationTest extends AuthenticatedIntegrationTestBase {
+
+  @Autowired private UserRepository userRepository;
+  @MockBean private PasswordEncoder passwordEncoder;
+  private User testUser;
+  private final AuthenticationRequestDto authRequest = new AuthenticationRequestDto();
+  private final HttpHeaders headers = new HttpHeaders();
+
+  @Override
+  @BeforeEach
+  void setup() {
+    userRepository.deleteAll();
+    createAndSaveTestUser();
+  }
+
+  @Test
+  void generateTokenSuccessfully() {
+
+    setupAuthRequestAndHeaders();
+    HttpEntity<AuthenticationRequestDto> requestEntity = new HttpEntity<>(authRequest, headers);
+
+    ResponseEntity<UserAuthDetailsDto> response =
+        testRestTemplate.exchange(
+            getBaseAuthUrl(), HttpMethod.POST, requestEntity, UserAuthDetailsDto.class);
+
+    assertEquals(HttpStatus.CREATED, response.getStatusCode());
+  }
+
+  @Test
+  void testUnauthorizedAccessWithInvalidToken() {
+    String invalidToken = "invalid_token";
+    addTokenToHeaders(invalidToken);
+    HttpEntity<Void> requestEntity = new HttpEntity<>(headers);
+    ResponseEntity<Void> response =
+        testRestTemplate.exchange(getBaseUserUrl(), HttpMethod.GET, requestEntity, Void.class);
+
+    assertEquals(HttpStatus.UNAUTHORIZED, response.getStatusCode());
+  }
+
+  @Test
+  void testAccessToProtectedEndpointWithValidToken() {
```

Comment:
Same here - fetchUsersWithValidTokenWillSucceed

Reply:
BTW, is this different than the successful case in the Users integration test in this case?

Reply:
You are correct. Did not realize it will duplicate old test. Removed it.